### ISSUE TITLE
Microsoft.OData.Client request failure when calling a function requiring array parameter

### ISSUE THREAD

**Issue by jamesra** [OData/odata.net#725]
I am using Microsoft.OData.Client to call a function requiring a collection of integers as a parameter.  The client and server appear to disagree how the array should be represented in the request's URI.

I first asked Stack Overflow with this issue without luck, but here is a link in case an answer shows up later:
[http://stackoverflow.com/questions/40687528/microsoft-odata-client-request-failure-when-calling-a-function-requiring-array-p](url)

### Assemblies affected
Microsoft.OData.Client v6.15 (Client)
Microsoft.OData.Core v7.0 (Server)

### Reproduce steps
My service defines a function that is not bound to a collection: ODataModelBuilder builder;

```
/*Removed unrelated model definitions…*/
var NetworkCellsFuncConfig = builder.Function("NetworkCells");
NetworkCellsFuncConfig.CollectionParameter<long>("IDs");
NetworkCellsFuncConfig.Parameter<int>("Hops");
NetworkCellsFuncConfig.ReturnsCollectionFromEntitySet<Structure>("Structures");
NetworkCellsFuncConfig.Namespace = null;
```
In my controller I define this function:
```

[HttpGet]
[EnableQuery(PageSize = 2048)]
[ODataRoute("NetworkCells(IDs={IDs},Hops={Hops})")]
public IQueryable<Structure> GetNetworkCells([FromODataUri] ICollection<long> IDs, [FromODataUri] int Hops)
{
    Request.ODataProperties().Path = DefaultODataPathHandler().Parse(System.Web.HttpContext.Current.Request.Url.GetLeftPart(System.UriPartial.Path), "Structures", Request.GetRequestContainer()) 
    return db.SelectNetworkStructures(IDs, Hops);
}
```

### Expected result
If I call my service using the URL fragment below I get a correct response:

> …/OData/NetworkCells(IDs=@IDs,Hops=1)?@IDs=[476]

### Actual result
The response is 404 - not found.  When I call the function from the client using Microsoft.OData.Client v6.15 this URL fragment is generated:

> ../OData/NetworkCells(IDs=@IDs,Hops=1)?@IDs=["476"]

### Additional details
The difference is the quotations around “476”. Removing the quotes returns a valid response.

Below is the auto-generated client code from the template:
```
/// <summary>
/// There are no comments for NetworkCells in the schema.
/// </summary>
[global::Microsoft.OData.Client.OriginalNameAttribute("NetworkCells")]
public global::Microsoft.OData.Client.DataServiceQuery<global::ODataClient.ConnectomeDataModel.Structure> NetworkCells(global::System.Collections.Generic.ICollection<long> IDs, int Hops)
{
    return this.CreateFunctionQuery<global::ODataClient.ConnectomeDataModel.Structure>("", "NetworkCells", false, new global::Microsoft.OData.Client.UriOperationParameter("IDs", IDs),
                    new global::Microsoft.OData.Client.UriOperationParameter("Hops", Hops));
}
```

I'm not familiar enough with OData spec to know what the proper formatting of the URI is, but the client/server should agree unless this is my mistake somehow.

I am hoping that this is caused by the version mismatch, but even so they both support OData v4 so it seems the generated URI's should be compatible.

**Comment by paulodero** [comment]
This is beacause by default `Int64 ` and `Decimal `parameters are serialized in IEEE754-compatible format. to avoid this, you need to pass in `IEEE754-compatible=false` as part of `Content-Type` header.

**Comment by jamesra** [comment]
Thank you!  I haven’t touched this code in a while but it is still in use.  I’ll refresh my memory and clean up whatever my workaround was.

**Comment by paulodero** [comment]
This option to pass the header parameter will however be available in the next release, probably `7.6.5`
